### PR TITLE
Adjust modifiers and add some properties to RaidenChannel & RaidenTransfer interfaces

### LINKED ISSUE #1657: Channel identifier can be undefined by interface definition
## Description

In PR #1636 it has turned out that the `id` field of the interface for
a `RaidenChannel` can be undefined. Due to out meeting and Andre [his
comment](https://github.com/raiden-network/light-client/pull/1636#discussion_r436844718) this is a due to the history of the SDK and is not the case anymore. So in fact the identifier is never undefined, but Typescript can't know. This should be simply fixed.

## Acceptance criteria
- `RaidenChannel.id` can be never `undefined`

## Tasks
- [ ] remove optionality of `id` field in interface definition of SDK
- [ ] remove unnecessary handling of undefined id in cli

### LINKED ISSUE #1708: Expose transfer secret on SDK
## Description

The [Raiden API for payments](https://raiden-network.readthedocs.io/en/latest/rest_api.html) (be aware that SDK calls payments transfers) defines the output format with the `secret` property. This property is simply not exposed by the `RaidenTransfer` object at the SDK. Therefor the `cli` can not access.

## Acceptance criteria
- `RaidenTransfer`  has the `secret` property
- the `cli` adopts to access this value on transfer to payment formatting

## Tasks
- [ ]

### PR DESCRIPTION
**Pull request by andrevmatos**
Fixes #1657 
Fixes #1708 

**Short description**
`RaidenChannel`'s `id`, `settleTimeout` and `openBlock` are marked as required and therefore always defined
`RaidenTransfer`'s `secret` is added as optional property, and'll hold transfer's secret whenever we know it (usually true for sent transfers)

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Nothing to test, mostly types and public data adjustments

### CODE REVIEW

File: raiden-cli/src/utils/channels.ts [hunk 1]
```diff
@@ -26,17 +26,16 @@ export function filterChannels(
   return filteredChannels;
 }
 
-/* eslint-disable @typescript-eslint/camelcase */
```
Comment (weilbith) [inline raiden-cli/src/utils/channels.ts:29]: Is this related to the new typescript-eslint rules? Or why can you remove that?

**Comment by weilbith** [comment]
If we do this quick. I can even integrate this into #1711 already and remove the issue reference.

File: raiden-cli/src/utils/channels.ts (hunk 1 above)
Comment (andrevmatos) [reply raiden-cli/src/utils/channels.ts:29]: Yes. Linting actually complains on this line on my machine freshly `npm ci`ed due to this rule not existing anymore. Check https://github.com/raiden-network/light-client/pull/1612#discussion_r433815089 for details
